plugins/services/src/js/components/ServiceStatusProgressBar.js

```diff
@@ -0,0 +1,88 @@
+import React, { PropTypes } from "react";
+import { Tooltip } from "reactjs-components";
+
+import StringUtil from "#SRC/js/utils/StringUtil";
+import ProgressBar from "#SRC/js/components/ProgressBar";
+import Pod from "../structs/Pod";
+import Service from "../structs/Service";
+import ServiceTree from "../structs/ServiceTree";
+import ServiceStatus from "../constants/ServiceStatus";
+
+class ServiceStatusProgressBar extends React.Component {
+  getTooltipContent() {
+    const { service } = this.props;
+    const { tasksRunning } = service.getTasksSummary();
+    const instancesTotal = service.getInstancesCount();
+
+    return (
+      <div className="tooltip-line-item">
+        <span>
+          {`${tasksRunning} ${StringUtil.pluralize("instance", tasksRunning)} running out of ${instancesTotal} `}
+        </span>
+      </div>
+    );
+  }
+
+  getProgressBar(data) {
+    const { service } = this.props;
+    const instancesCount = service.getInstancesCount();
+    const tasksRunning = service.getTaskCount();
+
+    if (tasksRunning === instancesCount) {
+      return (
+        <ProgressBar
+          className="status-bar--large"
+          data={data}
+          total={instancesCount}
+        />
+      );
+    }
+
+    return (
+      <Tooltip interactive={true} content={this.getTooltipContent()}>
+        <ProgressBar
+          className="status-bar--large"
+          data={data}
+          total={instancesCount}
+        />
+      </Tooltip>
+    );
+  }
+
+  render() {
+    const { service } = this.props;
+    const instancesCount = service.getInstancesCount();
+    const serviceStatus = service.getStatus();
+    const tasksRunning = service.getTaskCount();
+
+    if (instancesCount !== tasksRunning) {
+      return this.getProgressBar([
```

Comment:
Curious why you chose to split some of the rendering logic into `getProgressBar`?

Perhaps I'm missing something here but I see the logic could be much simpler if it was all contained in a single method as we have some duplication across the two - mostly same `const`s and same equality checks like `tasksRunning === instancesCount` (or it's inverse) and data is passed from one to the other that doesn't seem necessary.

Reply:
At first implementation/solution I ended up with that but I have refactored. thanks.